app/controllers/homepage_controller.rb

```diff
@@ -45,8 +45,38 @@ def publish
     redirect_to show_popular_links_path
   end
 
+  def destroy
+    if @latest_popular_links.can_delete?
+      @latest_popular_links.delete ? flash[:success] = "Popular links draft deleted.".html_safe : flash[:danger] = application_error_message
+    else
+      flash[:danger] = delete_published_error_message
+    end
+  rescue StandardError => e
+    Rails.logger.error "Error #{e.class} #{e.message}"
+    flash[:danger] = application_error_message
+  ensure
+    redirect_to show_popular_links_path
+  end
+
+  def confirm_destroy
+    if @latest_popular_links.can_delete?
+      render "homepage/popular_links/confirm_destroy"
+    else
+      flash[:danger] = delete_published_error_message
+      redirect_to show_popular_links_path
+    end
+  end
+
 private
 
+  def delete_published_error_message
```

Comment:
I think we can rename this to 'cannot_delete_published_record_error_message'. I know its bit long but I think the use of word 'cannot' is bit important.